lib/handlers/received_invoice_notification.dart

```diff
@@ -35,56 +35,65 @@ class InvoiceNotificationsHandler {
   _listenPaymentRequests() {
     // show payment request dialog for decoded requests
     _receivedInvoicesStream.where((payreq) => payreq != null && !_handlingRequest).listen((payreq) async {
-      final navigator = Navigator.of(_context);
-      var account = await _accountBloc.accountStream.firstWhere((a) => !a.initial, orElse: () => null);
-      if (account == null || !account.connected) {
```

Comment:
What is the reason for skip the connected condition?

Reply:
@roeierez I'll have to dig into this to make sure but I believe there's a race condition, or sth quite normal and `AccountModel` is assembled in parts. The first AccountModel that was not `initial`, wasn't necessarily `connected` and the invoice got skipped. The user had no feedback as to why what happened to the invoice on their clipboard.

I don't know which would be the right approach in terms of UX:

1. **Wait for account to be connected that's coming from `accountStream`**
   - `_accountBloc.accountStream.firstWhere((a) => !a.initial && a.connected,...`
   - This has the downside of payment request dialog showing much later, the user may forget what the payment was for, it could interrupt their current operations etc.

2. **Display the payment request right away** 
  - hope account is connected by the time it's paid or let it fail
  - either way, the user does not wonder what happened to the invoice on their clipboard and can retry it via "Paste Invoice or ID"
  
but I've opted for 2 as the feedback to the user is much more clear.